src/main/scala/performanceanalysis/DateTimeParser.scala

```diff
@@ -0,0 +1,40 @@
+package performanceanalysis
+
+import java.time.LocalDateTime
+
+import performanceanalysis.DateTimeParser._
+
+import scala.util.matching.Regex.Match
+import scala.util.{Failure, Success, Try}
+
+object DateTimeParser {
+
+  private val regex = """(\d{1,2}|\d{4})[ -/\.](\d{1,2})[ -/\.](\d{1,2}|\d{4})[ T]( \d|\d{2})[:.](\d{2})[.:](\d{2})(\.\d{0,9})?""".r
```

Comment:
Personally I'm not a fan of regexes. We could use Scala parser combinators here, which are better readable in my opinion.